### PR TITLE
[onton-completeness-pt-2] Patch 15: TUI input history

### PR DESCRIPTION
**Pull request by subsetpark**
## Summary
- Add `History` submodule to `Tui_input` — a bounded circular buffer for input line history
- Supports up/down browsing, consecutive duplicate suppression, empty line filtering, and capacity overflow (oldest entries dropped)
- Includes 9 inline tests covering all edge cases

## Test plan
- [x] `dune runtest` passes (all 42 tests including 9 new History tests)
- [x] `dune build` clean with no warnings
- [x] Format check passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/tui_input.ml [hunk 1]
```diff
@@ -146,3 +146,149 @@ let%test "apply_move noop preserves" = apply_move ~count:5 ~selected:3 Noop = 3
 
 let%test "apply_move noop clamps stale selection" =
   apply_move ~count:3 ~selected:9 Noop = 2
+
+(** Input history for the text-mode prompt.
+
+    Stores previously entered lines and supports up/down navigation. The history
+    is bounded; oldest entries are dropped when the capacity is exceeded. *)
+module History = struct
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/tui_input.ml]: Fixed — wired History into `input_fiber`: Up/Down call `older`/`newer`, Enter calls `push`, Escape calls `reset_browse`. Draft input is saved before browsing and restored when returning to `At_fresh`.
